hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -1288,13 +1286,13 @@ public void executeOnEntriesInternal(EntryProcessor entryProcessor, Predicate pr
             Map<Integer, Object> results;
             if (predicate instanceof PartitionPredicate) {
                 PartitionPredicate partitionPredicate = (PartitionPredicate) predicate;
-                Data key = toData(partitionPredicate.getPartitionKey());
-                int partitionId = partitionService.getPartitionId(key);
+                PartitionIdSet partitionIds = new PartitionIdSet(partitionService.getPartitionCount());
```

Comment:
Can you please also do this change for client protocol related classes? For instance `MapExecuteWithPredicateMessageTask`, now this class is executed on a single partition independent of the provided partition set size?

Reply:
I think that's the only class that needs changing. The rest of the client code uses the partition id to determine which cluster member to invoke the request on. `getPartitionKey` selects a random key from the set of partition keys on `PartitionPredicate` so it should have the desired result of load balancing.